File: cpp/src/ev1/ut3_strings/ejercicios/pendientes/E06_ReemplazarEspacios_Pendiente.cpp
/*
OBJETIVO: Lee una cadena y reemplaza todos los espacios por guiones bajos.
Este ejercicio ahora se ejecuta desde un menu do-while con opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E06 Reemplazar espacios (pendiente) ===" << endl;
        cout << "1. Intentar ejercicio" << endl;
        cout << "2. Ver objetivo" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1") {
            cout << "Escribe una cadena: ";
            string texto;
            getline(cin, texto);

            cout << "Instrucciones:" << endl;
            cout << "- Recorre el string y cambia cada ' ' por '_'." << endl;
            cout << "- (No hay Replace de C#; el string en C++ se modifica in-place)." << endl;
            cout << "- Guarda el resultado en otra variable si quieres conservar el original." << endl;

            string resultado = "";

            // TODO: reemplaza los espacios del texto por guiones bajos.

            cout << "Resultado: " << resultado << endl;
        } else if (opcion == "2") {
            cout << "Objetivo: sustituir espacios por guiones bajos recorriendo el string." << endl;
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}

File: csharp/src/ev1/ut3_strings/ejercicios/pendientes/E06_ReemplazarEspacios_Pendiente.cs
/*
OBJETIVO: Lee una cadena y reemplaza todos los espacios por guiones bajos (Replace).
Este ejercicio ahora se ejecuta desde un menu do-while con opcion 0 para salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E06 Reemplazar espacios (pendiente) ===");
    Console.WriteLine("1. Intentar ejercicio");
    Console.WriteLine("2. Ver objetivo");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
            Console.Write("Escribe una cadena: ");
            string texto = Console.ReadLine() ?? "";

            Console.WriteLine("Instrucciones:");
            Console.WriteLine("- Usa Replace(\" \", \"_\").");
            Console.WriteLine("- Guarda el resultado en otra variable.");

            string resultado = "";

            // TODO: reemplaza los espacios del texto por guiones bajos.

            Console.WriteLine("Resultado: " + resultado);
            break;

        case "2":
            Console.WriteLine("Objetivo: sustituir espacios por guiones bajos usando Replace().");
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");
